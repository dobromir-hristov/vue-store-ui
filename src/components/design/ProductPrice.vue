<template>
  <div class="ProductPrice font-serif text-3xl">
    {{ price | money(currency) }}
  </div>
</template>

<script>

import { ProductPriceMixin } from '@/mixins/core/ProductPriceMixin'

/**
 * Shows the price for the current viewed product
 * @module ProductPrice
 */
export default {
  name: 'ProductPrice',
  mixins: [ProductPriceMixin]
}
</script>
