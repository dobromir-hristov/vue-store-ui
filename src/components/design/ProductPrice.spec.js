import ProductPrice from './ProductPrice'

/**
 * @param overrides
 * @returns {{config, wrapper: Wrapper<V> | Wrapper<Vue> | *}}
 */
const createWrapper = (overrides) => {
  let config = _merge(createComponentMocks(),
    {
      propsData: {
        price: 55,
        currency: 'EUR'
      }
    },
    overrides
  )
  const wrapper = mount(ProductPrice, config)
  return {
    config,
    wrapper
  }
}

describe('ProductPrice', () => {
  it('renders the ProductPrice', () => {
    const { wrapper } = createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
