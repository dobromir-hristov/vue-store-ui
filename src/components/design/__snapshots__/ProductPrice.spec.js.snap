// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPrice renders the ProductPrice 1`] = `
<div class="ProductPrice font-serif text-3xl">
  â‚¬55
</div>
`;
